File: 13.Roman to integer/romanToInt.js
var romanToInt = function(s) {
    const symbol = {
    I: 1,
    V: 5,
    X: 10,
    L: 50,
    C: 100,
    D: 500,
    M: 1000
  };
  let result = 0;
  for(let i = s.length-1;i>=0;i--){
    let curr = s[i];
    let next = s[i+1];
    if(symbol[curr]<symbol[next]){
      result -= symbol[curr] ;
    }
    else{
      result += symbol[curr] ;
    }
  }
  return result;
}

File: 13.Roman to integer/romanToInt.py
class Solution(object):
    def romanToInt(self, s):
        symbol = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        result = 0
        for i in range(len(s)-1, -1, -1):
            curr = s[i]
            next_char = s[i+1] if i+1 < len(s) else None
            if next_char and symbol[curr] < symbol[next_char]:
                result -= symbol[curr]
            else:
                result += symbol[curr]
        return result
